Add basic income variables

nikhilwoodruff opened the pull request:
Adds some basic income variables from taxdata

MaxGhenis commented on openfisca_us_data/datasets/cps/base_cps/model_input_variables.py:
```diff
@@ -22,7 +31,90 @@ class interest_income(Variable):
+        def formula(person, period, parameters):
+            return person("P_UI_IMPUTE", period)
+
+    input_variables = [
```
```suggestion
    return [
```

MaxGhenis commented on openfisca_us_data/datasets/cps/base_cps/model_input_variables.py:
```diff
@@ -22,7 +31,90 @@ class interest_income(Variable):
+        e02100,
+        divs,
+        rents,
+        # e01500,
```
similarly, add to an issue

MaxGhenis commented on openfisca_us_data/datasets/cps/base_cps/dataset.py:
```diff
@@ -12,7 +12,16 @@
+    "P_FRSE_VAL",
+    "P_DIV_VAL",
+    "P_RNT_VAL",
+    # "P_RTM_VAL",
```
could these go in an issue instead if they're not in the ASEC?

MaxGhenis commented on openfisca_us_data/datasets/cps/base_cps/survey_input_variables.py:
```diff
@@ -8,12 +8,77 @@ def get_CPS_variables():
+        label = u"Imputed unemployment insurance"
+        definition_period = YEAR
+
+    CPS_variables = [
```
```suggestion
    return [
```

nikhilwoodruff commented:
@MaxGhenis added your suggestions, and it seems that the missing variables are imputed by C-TAM (but we could just use reported values for now). `RTM_VAL` seems to be pensions and annuities, though I'm not completely sure on that. Also added an example test for this PR, will add more.

nikhilwoodruff commented:
@MaxGhenis added your suggestions, and it seems that the missing variables are imputed by C-TAM (but we could just use reported values for now). `RTM_VAL` seems to be pensions and annuities, though I'm not completely sure on that. Also added an example test for this PR, will add more.

nikhilwoodruff commented:
@MaxGhenis I've changed to the simplified pandas-like framework, and added unit tests for all variables added by the dataset so far, and some general compatibility tests.

MaxGhenis commented on openfisca_us_data/datasets/cps/cps.py:
```diff
@@ -0,0 +1,95 @@
+    cps["person_family_id"] = person.PH_SEQ * 10 + person.PF_SEQ
+
+    # Add weights
+    cps["person_weight"] = person.A_FNLWGT / 1e2
```
should spm unit weight go here too?

nikhilwoodruff replied:
Added

MaxGhenis commented on openfisca_us_data/datasets/cps/cps.py:
```diff
@@ -0,0 +1,99 @@
+        ]
+
+        add_ID_variables(cps, person, family, household)
+        add_basic_income_variables(cps, person)
```
Can you change the name of this function?

nikhilwoodruff replied:
Changed
